### DISCUSSION TITLE
[Idea] Remote registration with ssh keys

### DISCUSSION THREAD

**Discussion by nykroy** [glpi-project/glpi-agent#358] [Ideas]
### Your idea

I think is best solution use ssh keys for ssh Remote registration is that posible ?

**Comment by g-bougard** [comment]
Hi @nykroy 

yes, this is possible.

If you're using a windows GLPI Agent, you will probably need to use the latest nightly build.
On unix/linux, you can simply setup ssh access configuration in the `~/.ssh/config` of the agent environment as you can need it from the console. This way, you can manage to use such kind of remote url: `ssh://remote-host/`.

**Reply by nykroy** [reply]
I will try it thanks!!

**Reply by Megachip** [reply]
What about multiple/different private keys? So an option for the specific key would be awesome.

**Reply by g-bougard** [reply]
When using libssh2, glpi-agent tries all private keys for which it finds a public key. When using ssh, just use `~/.ssh/config` to set up the right key.

**Reply by Megachip** [reply]
What about passwort protected keys?

**Reply by g-bougard** [reply]
While using libssh2, just set password to the protected keys password. This will involve glpi-agent will try first password authentication and this will fail, but the password will be re-used for key files if required. The failing user/password authentication attempt will involve a failure in debug log, but this is not critical.

To be honest, it's probably better to secure the computer hosting glpi-agent so you can use dedicated key without key protection. Key protection is really required if you need to use a key on a system you don't fully trust. Here with glpi-agent you still have to put the password somewhere in a configuration... so you still trust that computer and this is IMHO the same thing than using unprotected keys.

**Comment by nykroy** [comment]
Hi, how can i connect using glpi-agent 1.5  from windows to linux but using private key and passphrase? 

i am trying with this command but it respond "Cant Authenticate"
glpi-agent --remote=ssh://user@198.18.70.70 --logger=stderr --tasks remoteinventory -f --no-ssl-check --ca-cert-file=C:\Users\xxxx\Desktop\director_key.ppk

**Reply by g-bougard** [reply]
Hi @nykroy 

first of all, don't use no-ssl-check & ca-cert-file options, they are not related to your concern. GLPI-Agent will look in a specific place for private keys to try.

Secondly, ppk is a format used by putty, this is not the more standard openssh key format. You can use `puttygen` to export your key in this more common format and the expected one for the library the agent is using.

Then on windows, you have to put the exported private key into the a `var/.ssh` subfolder of your glpi-agent installation folder. I'm not sure if you can export the key protecting it with a password as I didn't test this case. So you can export it without password and secure the file so only administrators can read it. If you export with a password protection, try to use this password a remote password like in:
```
glpi-agent --remote=ssh://user:key-password@198.18.70.70 --logger=stderr --tasks remoteinventory -f --debug --debug
```
